llvm/test/Analysis/ValueTracking/known-non-zero.ll

```diff
@@ -882,11 +882,7 @@ define i1 @usub_sat_nonzero(i8 %xx, i8 %yy, i8 %ind) {
 ; CHECK-LABEL: @usub_sat_nonzero(
 ; CHECK-NEXT:    [[Y_ULT_31:%.*]] = icmp ult i8 [[YY:%.*]], 31
 ; CHECK-NEXT:    call void @llvm.assume(i1 [[Y_ULT_31]])
-; CHECK-NEXT:    [[XO:%.*]] = or i8 [[XX:%.*]], 34
```

Comment:
It has been handled in InstCombine by converting the `usub.sat` into `sub nuw`: https://godbolt.org/z/he76W3K3v

I think this patch just duplicates the logic in InstCombine: https://github.com/llvm/llvm-project/blob/8461d901a770516cf2069fe3bce979a6f8fc8d76/llvm/lib/Transforms/InstCombine/InstCombineCalls.cpp#L2150-L2160

Reply:
Ah yes, good point. Actually, now that you mention it, I'm pretty sure that @goldsteinn already submitted a patch along these lines on phabricator and we reached the same conclusion there... Maybe we should leave a comment so we don't get a third implementation :)

Reply:
You are right, I forgot :)

I'll close this and push a comment